tests/pyccel/scripts/hope_benchmarks_decorators/fib.py

```diff
@@ -1,6 +1,6 @@
 from pyccel.decorators import types
 
-@types('int')
+@types('int', results='int')
```

Comment:
@saidctb I understand that recursive functions need the `results` type to be explicitly provided by the user. I then have two questions:
1. What happens if the user does **not** provide `results` for a recursive function?
2. What happens if the user provides `results` but the function is **not** recursive?

Do you detect the errors? If so, what exceptions do you raise? I think you need to write two unit tests to check the cases above, and verify that the expected behavior is obtained (see https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions).

Reply:
1. If the user doesn't provide us the return type, then we can't deduce the return type, we should improve this in the future so that we can deduce the return type automatically.
2. In the case of a non recursive function ,the return type provided by the user doesn't matter, so it will be ignored for the moment.
for the first case we don't raise a proper exception error , so yes we should add it , as for the second case it will work fine and i don't think we should raise an error

Reply:
If the results type is completely ignored, it might be a good idea to raise a warning even if we don't raise an error. It could lead to unexpected behaviour for the user (e.g. function returns an int but the user specified a real). It's not a big problem, but a warning could be useful so the user is aware

Reply:
In case 2 (user provides results for non-recursive function) I suggest that we simply check if the provided type is the same as the inferred type. If not we should raise an exception.